api/v1alpha1/utils/commons.go

```diff
@@ -153,18 +183,30 @@ func (kc *KubeAuthConfiguration) GetServiceAccountName() string {
 	return kc.ServiceAccount.Name
 }
 
+func (kc *KubeAuthConfiguration) getCacheKey() string {
+	return fmt.Sprintf("%s:%s:%s:%s", kc.ServiceAccount.Name, kc.Path, kc.Role, kc.Namespace)
+}
+
 func (kc *KubeAuthConfiguration) GetVaultClient(context context.Context, kubeNamespace string) (*vault.Client, error) {
 	log := log.FromContext(context)
+
+	vaultClient := vaultClientCache.Get(kc)
```

Comment:
add a check that the jwt token isn't expired, if it is remove it from the map and continue with creating a new one